tests/integration/test_private_repos.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+
+import pytest
+from git import Repo
+
+from . import utils
+
+
+@pytest.mark.parametrize("env_package", ["https", "ssh"])
+def test_private_repos(env_package, test_env, tmpdir):
+
+    env_data = utils.load_test_data("private_repo_packages.yaml")[env_package]
+    is_supported_env = any(
+        x in str(os.environ.get("JOB_NAME")) for x in test_env.get("private_repo_test_envs", [])
+    )
+    if not is_supported_env:
+        pytest.skip(
+            (
+                "This test is only executed in environments that "
+                "have been configured with the credentials needed "
+                "to access private repositories."
+            )
+        )
+
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout"))
+    payload = {
+        "repo": env_data["repo"],
+        "ref": env_data["ref"],
+        "pkg_managers": [],
+        "flags": ["include-git-dir"],
+    }
+
+    initial_response = client.create_new_request(payload=payload)
+    completed_response = client.wait_for_complete_request(initial_response)
+
+    utils.assert_properly_completed_response(completed_response)
+    assert completed_response.data["packages"] == []
+    assert completed_response.data["dependencies"] == []
+
+    client.download_and_extract_archive(completed_response.id, tmpdir)
+    source_path = tmpdir.join(f"download_{str(completed_response.id)}")
+    repo = Repo(os.path.join(source_path, "app"))
+    assert repo.head.commit.hexsha == env_data["ref"]
```

Comment:
additional idea: check that `repo.git.diff()` is empty after checking the expected ref => we're basically checking all the expected files

Reply:
Nice! Will add